tests/test_pangomarkup_formatter.py

```diff
@@ -0,0 +1,45 @@
+"""
+    Pygments Pango Markup formatter tests
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import os
+import re
+import tempfile
+from io import StringIO
+from os import path
+
+import pytest
+
+from pygments.formatters import PangoMarkupFormatter, NullFormatter
+from pygments.lexers import PythonLexer
+
+TESTDIR = path.dirname(path.abspath(__file__))
+TESTFILE = path.join(TESTDIR, 'test_pangomarkup_formatter.py')
+
+with open(TESTFILE, encoding='utf-8') as fp:
+    tokensource = list(PythonLexer().get_tokens(fp.read()))
+
+
+def test_correct_output():
+    pfmt = PangoMarkupFormatter(nowrap=True)
+    poutfile = StringIO()
+    pfmt.format(tokensource, poutfile)
+
+    nfmt = NullFormatter()
+    noutfile = StringIO()
+    nfmt.format(tokensource, noutfile)
+
+    stripped_markup = re.sub('</?(b|u|i)>', '', poutfile.getvalue())
+    stripped_markup = re.sub('<span[^>]+>', '', stripped_markup)
+    stripped_markup = re.sub('</span>', '', stripped_markup)
+
+    text = noutfile.getvalue()
```

Comment:
This test file doesn't tell you if the PangoMarkupFormatter did anything right. I'd suggest you pass in a one-line code snippet and check the whole output for correctness. In the current test I could simply return the input text and that would pass.

Reply:
Actually, returning just the input text (even wrapped between `<tt>` and `</tt>`) would fail the test, but I will change like you suggested, because that's indeed much better. (Unless the default styles are going to change one day.)

Reply:
They may change, but then it's on me to update this. You could use the BW style though to ensure that the colors remain fixed (because the BW style will not change.)

Reply:
I'd rather keep the default style, because otherwise no `<span fgcolor="..."></span>` comes into play. In order to avoid any problems in the future, I simply strip the various color codes.